apps/site/src/_pages/docs/4_spec/02_graph-module.mdx

```diff
@@ -1,303 +1,445 @@
 
-Blocks may also add `default` and `examples` keys to `block-metadata.json` to define default and example properties conforming to their schema. If more complex example data is required that involves specifying _other_ entities and links between them, which the block will make use of, a block package may also include an `example-graph.json`.
+Blocks may also add `default` and `examples` keys to `block-metadata.json` to define default and example properties conforming to their schema. If more complex example data is required that involves specifying _other_ entities and link entities between them, which the block will make use of, a block package may also include an `example-graph.json`.
 
 #### <Hidden>Block package </Hidden>Specification
 
 ##### Blocks
 
 A block using the graph module MAY expand `block-metadata.json` to specify:
 
-- `schema`: a path to a JSON file describing the **properties** the block accepts as part of the **block entity,** using [JSON Schema vocabulary](https://json-schema.org/). This file may be included in the block package, or hosted at a remote address. If the schema hosted at a remote address is subject to change, bear in mind that changes in the schema may invalidate other values in the block metadata (e.g. `default`) or the block’s functioning (e.g. if important constraints are removed).
-- `default`: an object, conforming to the block’s schema, representing the default data that applications should provide when creating a block, unless (a) the block can handle being provided no data when first instantiated or (b) `variants` is provided (see below)
-- `examples`: an array of objects conforming to the block's schema which provide examples of the data that can be passed to it
+- `schema`: the versioned URL of the **entity type** associated with the block entity. As per the specification, the schema hosted at the remote address should not be subject to change and versions of entity types should be treated as immutable, permanently available, records.
+- `default`: an **entity**, conforming to the block’s entity type, representing the default data that applications should provide when creating a block, unless &#40;a&#41; the block can handle being provided no data when first instantiated or &#40;b&#41; `variants` is provided &#40;see below&#41;
+- `examples`: an array of objects conforming to the block's entity type which provide examples of the data that can be passed to it
 - `variants`: an array of objects, each with a `name` and `properties`, and optionally `description`, `icon`, and `examples`. These objects represents different variants of the block that the user can create, where `properties` sets the starting properties. As a simple example, a ‘header’ block might have variants with the `name` ‘Heading 1’ and ‘Heading 2’, which start with `{ level: 1 }` and `{ level: 2 }` as `properties`, respectively
```

Comment:
@CiaranMn can you help me update this? I'm not entirely sure what it is

Reply:
I'd update `level` to versioned URIs, it's fine otherwise.

Reply:
Can we have `properties` though? Presumably the user needs to be able to set both `properties` and `linkData`?

Reply:
I realise now that setting `linkData` is problematic because you don't statically know the `entityId`.. hm, this might prove troublesome for blocks that have a required link. Something to revisit in the future

Reply:
```suggestion
- `variants`: an array of objects, each with a `name` and `properties`, and optionally `description`, `icon`, and `examples`. These objects represents different variants of the block that the user can create, where `properties` sets the starting properties. As a simple example, a ‘header’ block might have variants with the `name` ‘Heading 1’ and ‘Heading 2’, which start with `{ "https://blockprotocol.org/@alice/types/property-type/level/": 1 }` and `{ "https://blockprotocol.org/@alice/types/property-type/level/": 2 }` as `properties`, respectively
```